### PR TITLE
influxdb functionality added

### PR DESCRIPTION
**Pull request by snghshb**
-updated ruby script for using influxdb
-added tag parser for influxdb
-added option to select influxdb database
-8086 port added for @socket

### CODE REVIEW

**Comment by npwalker** [comment]
Seems like we'd need to rename the script if it's going to handle more than graphite now?

We should probably merge in https://github.com/puppetlabs/puppet-metrics-viewer/pull/17 and rebase this onto it.  You'll have to update your code to allow setting the port.

File: json2graphite.rb [hunk 1]
```diff
@@ -12,7 +12,12 @@ def initialize(host)
 
   def socket
     return @socket if @socket && !@socket.closed?
-    @socket = TCPSocket.new(@host, 2003)
+    if $options[:database] == 'influxdb'
+      @socket = TCPSocket.new(@host, 8086)
+      puts "8086"
```
Comment (npwalker) [inline json2graphite.rb:19]: this should probably be removed.

**Comment by Sharpie** [comment]
#17 might not be relevant to these changes unless InfluxDB can accept line protocol input directly over a socket. So far it looks like a HTTP POST is required --- to set the target database and units for the timestamps.

File: json2graphite.rb [hunk 2]
```diff
@@ -50,22 +55,30 @@ def parse_file(filename)
...
+          #sleep 0.0001
+          nc.write("#{line}\n")
+        else
+          puts(line)
+        end
       end
     end
   rescue Exception => e
     STDERR.puts "ERROR: #{filename}: #{e.message}"
+    #STDERR.puts "#{e.backtrace}"
```
Comment (Sharpie) [inline json2graphite.rb:84]: Should leave this uncommented or leave it out.

File: json2graphite.rb [hunk 3]
```diff
@@ -115,6 +128,17 @@ def error_name(str)
...
+def return_tag(a, n)
+  if a[n].is_a? String
+    return a[n]
+  else
+    if n > -1
+      return_tag(a, n-1)
+    else return "none"
+  end
+end
+end
```
Comment (Sharpie) [inline json2graphite.rb:143]: Indentation on these last two `end` statements should be fixed.

**Review by Sharpie — CHANGES_REQUESTED** [review]

**Comment by suckatrash** [comment]
I have a new PR I could submit that would fix most of the issues called out here. Not sure what to do about renaming the script or changing how the port is handled at this point.

https://github.com/suckatrash/puppet-metrics-viewer/commit/e4e5dd93700034b57c6a7ec32fab60e2be224a45

**Comment by reidmv** [comment]
From a design perspective, ideally we'd split this into two separate scripts. I don't see a benefit to combining them, especially as there's no benefit to supporting two backends and we're likely to just be moving to InfluxDB.

```
json2graphite.rb
json2influxdb.rb
```

**Comment by npwalker** [comment]
You really think influxdb will be more prevalent at customers than graphite?  IMO one of the benefits of this was that the dashboards could be exported and used with customers piping the data into graphite from the script.  

I've seen far more graphite installs than influxdb installs.  If we can support both then nevermind but idk how to make a dashboard that works for both.

**Comment by reidmv** [comment]
@npwalker I think that's a side conversation. My main feedback was "Graphite and InfluxDB are different enough we should have different JSON transformation scripts for them. Unix philosophy."

On that side conversation: I would need to have a better sense of how performant InfluxDB is for our use case before having a strong opinion. I know for a fact that Graphite is really shaky for us, unless/until we can find a better way of mass-inserting our data into it. If InfluxDB is BETTER, that's the argument for moving to it. If and only if. Happy to talk more offline.

**Comment by Sharpie** [comment]
Finally circling back to this after some months of distraction, including a few performance issues where the work done here was especially helpful. I think we should integrate the Influx work alongside Graphite as a single script. The reason being that we have a common JSON format for input and splitting this into two scripts would mean duplicating the code for ingesting and parsing that.

I'll pull the commits here down, add some extensions and submit a new PR with the results.

**Comment by Sharpie** [comment]
Closing this out in favor of PR #21 which includes these patches plus some additional changes.

**Pull request closed without merging**